jenkins/bootstrap.py

```diff
@@ -330,10 +330,18 @@ def cat(self, path, generation):
         return self.call(cmd, output=True)
 
 
-    def upload_artifacts(self, path, artifacts):
+    def upload_artifacts(self, gsutil, path, artifacts):
         """Upload artifacts to the specified path."""
         # Upload artifacts
         if os.path.isdir(artifacts):
+            if os.path.basename(path) == os.path.basename(artifacts):
+                try:
+                    gsutil.stat(path)
+                    # Success means remote path exists
+                    base = os.path.basename(path) + '/'
+                    path = path[:-len(base)]
```

Comment:
Yes, but you wouldn't enter the if condition because the basenames are not the same. We know that for GCS it is 'artifacts', while it is '_artifacts' locally.